[18.0][MIG] account_reconcile_model_oca: Migration to 18.0

xaviedoanhduy opened the pull request:
### change in v18

- `get_sql` has removed in https://github.com/odoo/odoo/commit/5c397d56eb33dc37a0b0a8d1e7d4c43583c95950
- `account_journal_payment_credit_account_id` has removed in https://github.com/odoo/odoo/commit/01b87f1230beac0568f4e3b1b76e547909506892

### depend on

- https://github.com/OCA/account-reconcile/pull/740

etobella committed:
[ADD] account_reconcile_model_oca

Commit:
[UPD] Update account_reconcile_model_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (2 of 2 strings)

Translation: account-reconcile-17.0/account-reconcile-17.0-account_reconcile_model_oca
Translate-URL: https://translation.odoo-community.org/projects/account-reconcile-17-0/account-reconcile-17-0-account_reconcile_model_oca/it/

pedrobaeza committed:
[FIX] *: pre-commit fixes

Due to latest copier template with all the checks.

OCA-git-bot committed:
[BOT] post-merge updates

xtanuiha committed:
Added translation using Weblate (Chinese (Simplified) (zh_CN))

xtanuiha committed:
Translated using Weblate (Chinese (Simplified) (zh_CN))

Currently translated at 100.0% (2 of 2 strings)

Translation: account-reconcile-17.0/account-reconcile-17.0-account_reconcile_model_oca
Translate-URL: https://translation.odoo-community.org/projects/account-reconcile-17-0/account-reconcile-17-0-account_reconcile_model_oca/zh_CN/

xaviedoanhduy committed:
[IMP] account_reconcile_model_oca: pre-commit auto fixes

xaviedoanhduy committed:
[MIG] account_reconcile_model_oca: Migration to 18.0

mmequignon commented on account_reconcile_model_oca/models/account_bank_statement_line.py:
```diff
@@ -66,35 +65,45 @@ def _retrieve_partner(self):
-                   AS chunk
-                ))
-            """
+                SQL(
+                    rf"""
+                    {unaccent("%s")} ~* ('^' || (
+                        SELECT STRING_AGG(CONCAT('(?=.*\m', chunk[1], '\M)'), '')
+                        FROM regexp_matches({unaccent('partner.name')}, '\w{{3,}}', 'g')
+                        AS chunk
+                    ))
+                    """,
+                    text_value,
+                )
```
Why not adding the wrapping parenthesis here, in order to avoid the ugly extra parenthesis in the query parts every where else:
* `query = "[...] ("`
* `final = ") [...]"`
* `parts = ") OR (".join(subqueries)`

which IMO is more complicated to read/understand.

mmequignon reviewed (approved):
I'm not sure this is the right moment for cosmetic improvements.
LG though

TDu commented:
Are we not missing some commits from 17.0 https://github.com/OCA/account-reconcile/commits/17.0/account_reconcile_oca ?

xaviedoanhduy commented:
> Are we not missing some commits from 17.0 https://github.com/OCA/account-reconcile/commits/17.0/account_reconcile_oca ?

hi @TDu, Maybe you a little confused about between `account_reconcile_model_oca` and `account_reconcile_oca`?

nilshamerlinck commented on account_reconcile_model_oca/models/account_bank_statement_line.py:
```diff
@@ -0,0 +1,140 @@
+                query_parts,
+                self.company_id.id,
+            )
+            self._cr.execute(final_query.code, final_query.params)
```
```suggestion
            self._cr.execute(final_query)
```

nilshamerlinck replied:
see https://github.com/odoo/odoo/blob/c03fa022439cd48c18c929c64f829db9023c7a94/odoo/sql_db.py#L345

xaviedoanhduy committed:
[IMP] account_reconcile_model_oca: excludes account_accountant module

victoralmau committed:
[IMP] account_reconcile_model_oca: Create _get_write_off_move_move_line_dict() method to allow extension

FWP from 16.0: https://github.com/odoo/odoo/pull/188808

Related to OCA/account-reconcile#761

TT51885

victoralmau commented:
Please, cherry-pick https://github.com/OCA/account-reconcile/pull/790 to commit history

victoralmau commented:
Please, cherry-pick https://github.com/OCA/account-reconcile/pull/804 to commit history.

xaviedoanhduy commented:
> Please, cherry-pick #804 to commit history.

Hi @victoralmau, I'm a bit confused about your PR affecting `account_reconcile_oca` and how it relates to this module (`account_reconcile_model_oca`)?

victoralmau commented:
> > Please, cherry-pick #804 to commit history.
> 
> Hi @victoralmau, I'm a bit confused about your PR affecting `account_reconcile_oca` and how it relates to this module (`account_reconcile_model_oca`)?

Oops sorry, I got the module mixed up, you're right, there's nothing to add.

JordiBForgeFlow reviewed (approved):
:+1: Functional review

JordiBForgeFlow commented:
/ocabot merge nobump

pedrobaeza commented:
/ocabot migration account_reconcile_model_oca
